tensorflow_datasets/image/coil100.py

```diff
@@ -85,10 +87,12 @@ def _generate_examples(self, data_dir_path):
     for file_name in tf.io.gfile.listdir(data_dir_path):
       if file_name.endswith(".png"):
         image = os.path.join(data_dir_path, file_name)
-        label = file_name.split("_")[2].split(".")[0]
+        angle_label = file_name.split("_")[2].split(".")[0]
+        angle = tf.cast(int(angle_label), tf.int64)
```

Comment:
Did you actually tested this code ? _generate_examples should yield np.array compatible values, not tf.Tensor.

Reply:
I tested the updated dataset with `load` and it seem to be yielding each example as a `Dict`. 
The signature of each example is as -
```
{
  'angle': <tf.Tensor 'args_0:0' shape=() dtype=int64>, 
  'angle_label': <tf.Tensor 'args_1:0' shape=() dtype=int64>, 
  'image': <tf.Tensor 'args_2:0' shape=(128, 128, 3) dtype=uint8>, 
  'object_id': <tf.Tensor 'args_3:0' shape=() dtype=int64>
}
```

Thank you.

Reply:
Ok! I think you mean instead of `tf.cast` we should use `numpy`conversion for `np.array` compatibility.